src/lib/components/UrlChooser/index.js

```diff
@@ -0,0 +1,158 @@
+    // non-properties (stolen DOM nodes)
+    this._urlInput = null;
+    this._runLighthouseButton = null;
+  }
+
+  render() {
+    return html`
+      <div class="report_header_enterurl">
+        <div class="lh-enterurl lh-enterurl--selected">${this.url}</div>
+        <div class="lh-enterurl lh-enterurl--switch">
+          <input
+            type="url"
+            class="lh-input"
+            placeholder="Enter a web page URL"
+            pattern="https?://.*"
+            minlength="7"
+            @keyup="${this.onUrlKeyup}"
+          />
+          <button
+            class="lh-enterurl__close"
+            aria-label="Remove URL"
+            @click=${this.clearInput}
+          ></button>
+        </div>
+        <div class="lh-controls">
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--secondary"
+            @click=${this.switchUrl}
+          >
+            Switch URL
+          </button>
+          <button
+            ?disabled=${this.active}
+            class="w-button w-button--primary"
+            id="run-lh-button"
+            @click=${this.requestAudit}
+          >
+            Run Audit
+          </button>
+        </div>
+      </div>
+    `;
+  }
+
+  updated(changedProperties) {
+    this._urlInput = this.renderRoot.querySelector('input[type="url"]');
+    this._runLighthouseButton = this.renderRoot.querySelector("#run-lh-button");
+
+    if (!changedProperties.has("url")) {
+      return;
+    }
+
+    const inputValue = this._urlInput.value;
+    const updateUrl = this.url;
+
+    if (this.switching && updateUrl && !inputValue) {
+      // if the user has just signed in, the element was in an initial state,
+      // AND the user hasn't typed anything, reset element with URL
+      this._urlInput.value = updateUrl;
+      this.switching = false;
+    } else if (updateUrl == null && !this.switching) {
+      // if the user has signed out, clear the href and enter switching mode
+      this._urlInput.value = null;
+      this.switching = true;
+    } else if (!this.switching) {
+      // in all other cases, only update the URL if ther user isn't switching
+      this._urlInput.value = updateUrl;
+    }
+  }
+
+  requestAudit() {
+    let url = this.url;
+
+    if (this.switching) {
+      this.fixUpUrl();
+
+      if (!this._urlInput.validity.valid) {
+        const detail = `Invalid URL. Please enter a full URL starting with https://.`;
+        const event = new CustomEvent("web-error", {bubbles: true, detail});
+        this.dispatchEvent(event);
+        return;
+      }
+
+      // if Request was pressed as part of switching, "url" still reflects prior state, so request
+      // audit based on the typed URL
+      this.switching = false;
+      url = this._urlInput.value;
+    }
+
+    const event = new CustomEvent("audit", {detail: url});
+    this.dispatchEvent(event);
+  }
+
+  switchUrl() {
+    this.switching = true;
+
+    const input = this._urlInput;
```

Comment:
I've done something different. I've realised that the awkward `requestAnimationFrame` is actually because we haven't rendered yet.

So I've added a `updated()` handler which focuses on the `urlInput` if `switching` is now true, and made the lit-html template just set `switching = true` as an inline handler.